File: src/server/auth/users.repository.ts
import { ensureSchema, getDb } from '@/server/shared/db/client';

export async function insertUser(
  userId: string,
  username: string,
  passwordHash: string,
  createdAt: number
): Promise<void> {
  await ensureSchema();
  await getDb().execute({
    sql: 'INSERT INTO users (id, username, password_hash, created_at) VALUES (?, ?, ?, ?)',
    args: [userId, username, passwordHash, createdAt],
  });
}

export async function findUserByUsername(
  username: string
): Promise<{ id: string; passwordHash: string } | null> {
  await ensureSchema();
  const result = await getDb().execute({
    sql: 'SELECT id, password_hash FROM users WHERE username = ?',
    args: [username],
  });
  const row = result.rows[0];

  if (!row) return null;

  return {
    id: String(row.id),
    passwordHash: String(row.password_hash),
  };
}

File: backend/app/auth/users_repository.py
from app.shared import database


def insert_user(
    user_id: str, username: str, password_hash: str, created_at: int
) -> None:
    database.ensure_schema()
    database.get_client().execute(
        "INSERT INTO accounts (id, username, password_hash, created_at) VALUES (?, ?, ?, ?)",
        [user_id, username, password_hash, created_at],
    )


def find_user_by_username(username: str) -> dict[str, str] | None:
    database.ensure_schema()
    result = database.get_client().execute(
        "SELECT id, password_hash FROM accounts WHERE username = ?",
        [username],
    )
    row = result.rows[0] if result.rows else None

    if not row:
        return None

    return {
        "id": str(row["id"]),
        "password_hash": str(row["password_hash"]),
    }
